src/fsharp/AugmentWithHashCompare.fs

```diff
@@ -868,25 +871,29 @@ let slotImplMethod (final, c, slotsig) : ValMemberInfo =
     IsImplemented=false
     ApparentEnclosingEntity=c} 
 
-let nonVirtualMethod c : ValMemberInfo = 
+let nonVirtualMethod mk c : ValMemberInfo =
   { ImplementedSlotSigs=[]
     MemberFlags={ IsInstance=true 
                   IsDispatchSlot=false
                   IsFinal=false
                   IsOverrideOrExplicitImpl=false
-                  MemberKind=SynMemberKind.Member}
+                  MemberKind=mk}
     IsImplemented=false
     ApparentEnclosingEntity=c} 
 
 let unitArg = ValReprInfo.unitArgData
 let unaryArg = [ ValReprInfo.unnamedTopArg ]
 let tupArg = [ [ ValReprInfo.unnamedTopArg1; ValReprInfo.unnamedTopArg1 ] ]
-let mkValSpec g (tcref: TyconRef) tmty vis slotsig methn ty argData = 
+let mkValSpec g (tcref: TyconRef) tmty vis slotsig (methn: string) ty argData =
     let m = tcref.Range 
     let tps = tcref.Typars m
     let membInfo = 
         match slotsig with 
-        | None -> nonVirtualMethod tcref 
+        | None ->
+            let mk =
+                if methn.StartsWith("get_", System.StringComparison.Ordinal) then SynMemberKind.PropertyGet
```

Comment:
Minor nit: I'd prefer you pass in a boolean flag rather than relying on the prefix here

Reply:
Done 👍